flecsi/execution/hpx/task_epilog.h

```diff
@@ -0,0 +1,280 @@
+/*
+    @@@@@@@@  @@           @@@@@@   @@@@@@@@ @@
+   /@@/////  /@@          @@////@@ @@////// /@@
+   /@@       /@@  @@@@@  @@    // /@@       /@@
+   /@@@@@@@  /@@ @@///@@/@@       /@@@@@@@@@/@@
+   /@@////   /@@/@@@@@@@/@@       ////////@@/@@
+   /@@       /@@/@@//// //@@    @@       /@@/@@
+   /@@       @@@//@@@@@@ //@@@@@@  @@@@@@@@ /@@
+   //       ///  //////   //////  ////////  //
+
+   Copyright (c) 2016, Los Alamos National Security, LLC
+   All rights reserved.
+                                                                              */
+#pragma once
+
+/*! @file */
+
+#include <flecsi/data/dense_accessor.h>
+#include <flecsi/data/ragged_accessor.h>
+#include <flecsi/data/ragged_mutator.h>
+#include <flecsi/data/sparse_accessor.h>
+#include <flecsi/data/sparse_mutator.h>
+
+/*!
+ @file
+ @date Initial file creation: May 19, 2017
+ */
+
+#include <vector>
+
+#include "mpi.h"
+#include <flecsi/coloring/mpi_utils.h>
+#include <flecsi/data/data.h>
+#include <flecsi/data/dense_accessor.h>
+#include <flecsi/execution/context.h>
+
+namespace flecsi {
+namespace execution {
+
+/*!
+ The task_epilog_t type can be called to walk the task args after the
+ task has run. This allows synchronization dependencies to be added
+ to the execution flow.
+
+ @ingroup execution
+ */
+
+struct task_epilog_t : public utils::tuple_walker__<task_epilog_t> {
+
+  /*!
+   Construct a task_epilog_t instance.
+   */
+
+  task_epilog_t() = default;
+
+  /*!
+   FIXME: Need a description.
+  





```

Comment:
This file has been converted to CRLF line endings (after its creation, in 786c6053)&mdash;and these lines have been given _additional_ carriage returns.  Amusingly, `clang-format` is not idempotent for them: it adds another `\r` every time and never converges.

Reply:
Uhh, good catch, I'll fix this.

Reply:
Maybe this doesn't matter, but shouldn't _all_ of the CRs have gone for consistency with other files, not just the extra ones?

Reply:
Grrr. I'm not sure what's happening here. I think we will remove those files anyways as they are plain copies of the corresponding files from the MPI backend. But I will have a closer look.